code/modules/mob/living/silicon/robot/inventory.dm

```diff
@@ -51,6 +51,8 @@
 		module_state_3:loc = module
 		module_state_3 = null
 		//inv3.icon_state = "inv3"
+	for (var/obj/screen/inv in src.HUDinventory)
+		inv.update_icon()
```

Comment:
```suggestion
	for(var/obj/screen/inv as anything in HUDinventory)
		inv.update_icon()
```

Reply:
🤔
Not as versed in byond optimization. Performance improvement by just taking anything it gives me, yeah?

Reply:
For typed loops BYOND will do an `istype()` check, and will skip a list entry if it's type does not match an iterator's type.
But if we know for sure that list only contains datums of a specific type (which it does in our case) - we can skip those type checks with `as anything` keyword.
It's not much in the way of performance gain, but I don't see why neglect it.
As for `src.`, it's simply redundant and should be avoided as per Eris code contribution guidelines.

Reply:
Ah, I see. Didn't know it'll skip the type check with `as anything`
Interesting, definitely going to watch for that in the future.